eventually-postgres/src/subscriber.rs

```diff
@@ -23,20 +23,26 @@ use tokio_postgres::AsyncMessage;
 
 const DEFAULT_BROADCAST_CHANNEL_SIZE: usize = 128;
 
-/// Alias type for a `Result` having [`DeserializeError`] as error type.alloc
+/// Alias type for a `Result` having [`SubscriberError`] as error type.alloc
 ///
-/// [`DeserializeError`]: struct.DeserializeError.html
-pub type Result<T> = std::result::Result<T, DeserializeError>;
+/// [`SubscriberError`]: struct.SubscriberError.html
+pub type Result<T> = std::result::Result<T, SubscriberError>;
 
 /// Error returned by the `TryStream` on [`subscribe_all`]
-/// when deserializing payloads coming from Postgres' `LISTEN`
-/// asynchronous notifications.
 ///
 /// [`subscribe_all`]: struct.EventSubscriber.html#method.subscribe_all
 #[derive(Debug, Clone, thiserror::Error)]
-#[error("failed to deserialize notification payload from JSON: {message}")]
-pub struct DeserializeError {
-    message: String,
+pub enum SubscriberError {
```

Comment:
If we have `SubscriberError`, we can remove the duplicated "Error" part from the enum variants:

```rust
pub enum SubscriberError {
    Deserialize(String),
    Connection(tokio_postgres::Error),
}
```